product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -114,3 +103,30 @@ def price_compute(self, price_type, uom=False, currency=False, company=False):
         return super(ProductProduct, self).price_compute(
             price_type, uom, currency, company
         )
+
+    def _prepare_customerinfo(self, params):
+        partner_id = params.get("partner_id")
+        return self.env["product.customerinfo"].search(
+            [
+                ("name", "=", partner_id),
+                "|",
+                ("product_id", "=", self.id),
+                "&",
+                ("product_tmpl_id", "=", self.product_tmpl_id.id),
+                ("product_id", "=", False),
+            ],
+            order="product_id, sequence",
+        )
+
+    def _select_customerinfo(
+        self, partner=False, _quantity=0.0, _date=None, _uom_id=False, params=False
+    ):
+        """Customer version of the standard `_select_seller`. """
+        # TODO: For now it is just the function name with same arguments, but
+        #  can be changed in future migrations to be more in line Odoo
+        #  standard way to select supplierinfo's.
+        if not params:
+            params = dict()
+        params.update({"partner_id": partner.id})
+        res = self._prepare_customerinfo(params)
+        return res[:1]
```

Comment:
Maybe a limit=1 in the domain would suffice?

Reply:
I want to keep the structure like [_sellect_sellers](https://github.com/odoo/odoo/blob/14.0/addons/product/models/product.py#L602) as much as possible

Reply:
This is not adding anything useful, as the return result is going to be the same 1 record, but with less performance and readability. Not always the upstream code is the one to imitate...

Reply:
Ok, done.